### ISSUE TITLE
Radio buttons in table flagged as not being part of group

### ISSUE THREAD

**Issue by gerardkcohen** [dequelabs/axe-core#306]
The following example is flagged with [Rule Id: Radiogroup](https://dequeuniversity.com/rules/axe/1.1/radiogroup)

Ex:

```
        <table>
            <caption>Select a Product</caption>
            <thead>
                <tr>
                    <th scope="col">Select</th>
                    <th scope="col">Product</th>
                    <th scope="col">Description</th>
                    <th scope="col">Cost</th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <th scope="row">Basic Web Server</th>
                    <td>
                        <label>Select Basic
                            <input type="radio" name="product" value="basic" />
                        </label>
                    </td>
                    <td>Basic server to host your basic site</td>
                    <td>$199/yr</td>
                </tr>
                <tr>
                    <th scope="row">Awesome Web Server</th>
                    <td>
                        <label>Select Awesome
                            <input type="radio" name="product" value="awesome" />
                        </label>
                    </td>
                    <td>Awesome server to host your awesome site</td>
                    <td>$299/yr</td>
                </tr>
                <tr>
                    <th scope="row">SSL Cert</th>
                    <td>
                        <label>Select SSL Cert
                            <input type="radio" name="product" value="ssl" />
                        </label>
                    </td>
                    <td>Make sure your site is secure</td>
                    <td>$99/yr</td>
                </tr>
            </tbody>
        </table>

```
This should be valid use case, even though not wrapped in an explicit grouping/ form role. If not, a suggestion on how to remediate would be good.

Thanks!

**Comment by SiTaggart** [comment]
Ooooh yeah, I have exactly the same question / issue! +1 on this

**Comment by marcysutton** [comment]
@WilcoFiers @dylanb did you have any ideas for how to solve this? Adding `aria-labelledby` to the INPUT elements removes the violation, but it also stomps over the label text already provided. Perhaps we could allow `aria-describedby` to make the association with a table header? The table caption is announced in Voiceover for the first input, but not the rest. It isn't announced in JAWS or NVDA, at least in the versions I tested.

**Comment by WilcoFiers** [comment]
Different origin, but #384 should address this too.

**Comment by dylanb** [comment]
@marcysutton if you look at the example that originated this ticket, the table has a caption that is the group label and the radio buttons are the only inputs in the table.

This would be a case we could definitely pass.

Other cases might be:
1. Common `<th>` with text and the correct `scope`
1. Common header-id association
1. Common aria-labelledby

We might also consider `<hN>` if the header can be uniquely associated with this group i.e. minimal intervening content

**Comment by WilcoFiers** [comment]
@dylanb Shouldn't that be aria-describedby? The shared table heading thing I can do. Not sure how to do `<hN>` though. Can we pass on that for now?

**Comment by dylanb** [comment]
We could do `aria-describedby` but it is not actually part of the accessible name calculation thats why I thought to exclude that.

I agree we could defer `<hN>` too

**Comment by marcysutton** [comment]
When we talked about it just now Dylan reminded me that `aria-labelledby="id1 id2"` would expose both labels as the accessible name, while `aria-describedby` would not.

**Comment by WilcoFiers** [comment]
Fair point... which begs the question... what should we do about input elements that don't have a unique label?

**Comment by WilcoFiers** [comment]
This will be closed when #1640 is merged.
